packages/driver/src/cross-origin/patches/cookies.js

```diff
@@ -77,4 +136,30 @@ export const patchDocumentCookie = (Cypress, window) => {
   }
 
   window.addEventListener('unload', onUnload)
+
+  Cypress.on('window:before:load', patch)
+
+  Cypress.specBridgeCommunicator.on('cross:origin:cookies', (cookies) => {
```

Comment:
This updates the cookies in the cross origin aut with changes pushed from the primary cypress instance? This should be pretty easy to not rely on cypress for.

Reply:
This ultimately comes from the proxy. If the spec bridge is already set up, it sends this event to pass in the cookies to the patch before the page load that might potentially use them via document.cookie.